### PR TITLE
Windows build 

### PR DESCRIPTION
**Pull request by tjwilli58**
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Trying to get recipe modified to build on Windows.  Used bld.bat from gobject-introspection-feedstock and patterned changes on meta.yaml from there also.

Partially building. Not sure what's wrong yet though. All suggestions appreciated.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

**Comment by ccordoba12** [comment]
@conda-forge/core, a little help here please. Thanks!

File: recipe/bld.bat [hunk 1]
```diff
@@ -0,0 +1,38 @@
+@ECHO ON
+
+@REM I cannot for the life of me figure out how Cygwin/MSYS2 figures out its
```
Comment (mingwandroid) [inline recipe/bld.bat]: Cygwin/MSYS2 locate themselves entirely from the location of the {cygwin-1.0,msys-2.0}.dll

File: recipe/bld.bat [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@REM root directory, which it uses to find the /etc/fstab which *sometimes*
+@REM affects the choice of the cygdrive prefix. But, regardless of *why*,
+@REM I find that we need this to work:
+mkdir %BUILD_PREFIX%\Library\etc
+echo none / cygdrive binary,user 0 0 >%BUILD_PREFIX%\Library\etc\fstab
+echo none /tmp usertemp binary,posix=0 0 0 >>%BUILD_PREFIX%\Library\etc\fstab
```
Comment (mingwandroid) [inline recipe/bld.bat]: Instead of any of this, add a dependency of `m2-filesystem`.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -81,13 +96,14 @@ outputs:
         # expat here is _only_ required for mesa-dri-drivers
         - {{ cdt('expat') }}                 # [linux]
         - {{ cdt('libselinux-devel') }}      # [linux]
-        - pkg-config
-        - m4
-        - autoconf
-        - automake
+        - {{native}}pkg-config
```
Comment (mingwandroid) [inline recipe/meta.yaml]: m2w64-pkg-config is for finding mingw-w64 libraries, I am not sure that's what you want here, probably not.

Comment (tjwilli58) [reply recipe/meta.yaml]: I'll change to m2-pkg-config. Thanks.

File: recipe/bld.bat (hunk 2 above)
Comment (tjwilli58) [reply recipe/bld.bat]: Will do.

**Comment by mingwandroid** [comment]
You don't want any m* pkg-config.. You want one that looks in the msvc directories instead.

**Comment by tjwilli58** [comment]
> You don't want any m* pkg-config.. You want one that looks in the msvc directories instead.

Where do I get this then?


```
(base) C:\Users\Tim>conda search "*pkg-config*"
Loading channels: done
# Name                       Version           Build  Channel
m2-pkg-config                 0.29.1               1  pkgs/msys2
m2-pkg-config                 0.29.1               2  pkgs/msys2
m2w64-pkg-config              0.29.1               1  pkgs/msys2
m2w64-pkg-config              0.29.1               2  pkgs/msys2

```

**Comment by mingwandroid** [comment]
You may need to build it, is it definitely essential? I am not familiar with meson, how do you tell it where to find things and what compilers to use etc?

**Comment by tjwilli58** [comment]
I'm actually not very familiar with meson either (or building with msvc either for that matter - I'm mainly a Python programmer.) I'm just following the success @pkgw has had with building [gobject-introspection-feedstock](https://github.com/conda-forge/gobject-introspection-feedstock) and [glib-feedstock](https://github.com/conda-forge/gobject-introspection-feedstock) on Windows.

Re: this [PR](https://github.com/conda-forge/gobject-introspection-feedstock/pull/9)

**Comment by tjwilli58** [comment]
I put m2w64-pkg-config back instead of m2-pkg-config. That finds the libraries and things get compiled.
The error I'm finding is with `ninja test` in `bld.bat`. If I comment that out, gst-plugins-base gets build, but the build for the  gstreamer output fails.

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

**Comment by tjwilli58** [comment]
@conda-forge-admin, please rerender

**Comment by tjwilli58** [comment]
@conda-forge-admin, please rerender

**Comment by tjwilli58** [comment]
@conda-forge-admin, please rerender

**Comment by tjwilli58** [comment]
How do I avoid the win_c_compilervs2008 check?

**Comment by mariusvniekerk** [comment]
You can ignore it

**Comment by nmaxwell** [comment]
Can I ask what the status of this is? We're interested in running GStreamer on Windows, and are already using anaconda for everything. The current build status says the windows build is disabled. Thanks!

**Comment by pkgw** [comment]
@nmaxwell I think the status is about what you see here — there's been interest but no one's pushing this over the finish line.

**Comment by tjwilli58** [comment]
I'd love to get back to this, because I need it too, but I realized that I'm not the right person to be doing this. I forget what I got hung up on, but I kept running into problems building this.

I think that the build process for gstreamer has progressed a bit though, so it may be easier. 
For now, I'm using [msys2](https://www.msys2.org/) for gstreamer and gst-python.

**Comment by jakirkham** [comment]
cc @seanyen (in case this is of interest to you 😉)

**Pull request closed without merging**